docs/pycanon/pycanon_and_privacy_metrics.md

```diff
@@ -0,0 +1,115 @@
+
+### T-Closeness
+
+A database with one sensitive attribute S verifies t-closeness if all the equivalence classes verify it. An equivalence class verifies t-closeness if the distribution of the values of S are at a distance no closer than t from the distribution of the sensitive attribute in the whole database.
+
+In PyCanon, [Earth Mover's distance](https://ieeexplore.ieee.org/document/4221659) is used. See this paper and the PyCanon [paper](https://www.nature.com/articles/s41597-022-01894-2) for more information.
+
+### δ-Disclosure Privacy
+
+For a dataset with only one sensitive attribute S, the δ-disclosure privacy is verified if:
+
+$$
+\log\left(\frac{p(EC,S)}{p(D,S)}\right) < \delta_s,
+$$
+
+for every sensitive attribute in the dataset, and every equivalence class. p(EC,s) is the fraction of records with s as a sensitive attribute in the equivalence class EC, and D is the dataset.
+
+## Other Metrics
+
+There are various other methods for measuring privacy which are not included in the PyCanon tool. They are well explained in these [blogs](https://desfontain.es/blog/). We will discuss [K-Map](https://desfontain.es/blog/k-map.html) alongside [δ-presence](https://desfontain.es/blog/delta-presence.html). 
+
+### K-Map
+
+Like K-Anonymity, K-Map requires a list of quasi-identifiers. It also requires a larger re-identification dataset. This could be an original dataset before anonymisation, or another larger dataset of everyone living in the UK, for example a census. Your data satisfies k-map if every combination of values for the quasi identifiers appears at least k times in the re-identification dataset. Calculating K-Map is very computationally expensive.
+
+### δ-Presence
+
+Similar to k-map, δ-presence is a metric that quantifies the probability that an individual belongs to an anonymised dataset. It is also very computationally expensive.
+
+
+### Visualisations
+
+Google Cloud's [Looker Studio](https://cloud.google.com/sensitive-data-protection/docs/visualizing_re-id_risk) has a visualisation suite for privacy metrics.
+
+## Anomoly Detection
+
+Our current privacy scoring method uses Named-Entity-Recognition (NER) to extract identifiers and quasi-identifiers, then [PyCorrectMatch](https://github.com/computationalprivacy/pycorrectmatch) measures the data uniqueness of each piece of text based on the extracted entities. Each row is given a score on how unique the data is. Unique pieces of data are assumed to have a higher risk of re-identification.
+
+Following this idea, we instead can explore the use of anomoly detection to identify high risk columns.
+
+### Distance Based Methods
+
+Distance metrics can be used to find outliers. The greater the distance away from other individuals, the more unique they are. Possible [distance metrics](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.DistanceMetric.html) that can be used are:
+
+- Euclidean Distance
+- Manhatten Distance
+- Minowski Distance
```

Comment:
Add any thoughts on where best to use each one?

Reply:
Added a couple of sentences:

Different distance metrics will have an effect on which data points are considered outliers. As our data is unlikely to be high-dimensional, simple methods such as Euclidean Distance can be both effective and explainable. However, for higher dimensional multi-variate data, more advanced distance metrics could be used, such as Mahalanobis which is effective for high dimensional multi-variate data.